BUG: Building with python<3.9 should error out early in the build process with a clear message

### Describe the issue:

I can not compile numpy from the source code in commit 1bb932f69e0974f11c98b6036529a014aa8dfaf2 . 



### Reproduce the code example:

```python
git clone git@github.com:numpy/numpy.git
git submodule update --init
python -m pip install -r test_requirements.txt
python runtests.py -v
```


### Error message:

```shell
error: Command "x86_64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -Werror=vla -Werror=nonnull -Werror=pointer-arith -Wlogical-op -Werror=unused-function -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DNPY_INTERNAL_BUILD=1 -DHAVE_NPY_CONFIG_H=1 -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE=1 -D_LARGEFILE64_SOURCE=1 -Ibuild/src.linux-x86_64-3.8/numpy/core/src/multiarray -Ibuild/src.linux-x86_64-3.8/numpy/core/src/common -Ibuild/src.linux-x86_64-3.8/numpy/core/src/umath -Inumpy/core/include -Ibuild/src.linux-x86_64-3.8/numpy/core/include/numpy -Ibuild/src.linux-x86_64-3.8/numpy/distutils/include -Ibuild/src.linux-x86_64-3.8/numpy/core/src/npysort -Inumpy/core/src/common -Inumpy/core/src -Inumpy/core -Inumpy/core/src/npymath -Inumpy/core/src/multiarray -Inumpy/core/src/umath -Inumpy/core/src/npysort -Inumpy/core/src/_simd -I/usr/include/python3.8 -Ibuild/src.linux-x86_64-3.8/numpy/core/src/common -Ibuild/src.linux-x86_64-3.8/numpy/core/src/npymath -c numpy/core/src/multiarray/arrayfunction_override.c -o build/temp.linux-x86_64-3.8/numpy/core/src/multiarray/arrayfunction_override.o -MMD -MF build/temp.linux-x86_64-3.8/numpy/core/src/multiarray/arrayfunction_override.o.d -msse -msse2 -msse3" failed with exit status 1

when I trace back:
it bloated at here. 
numpy/core/src/multiarray/arrayfunction_override.c: At top level:
numpy/core/src/multiarray/arrayfunction_override.c:693:40: error: ‘Py_TPFLAGS_HAVE_VECTORCALL’ undeclared here (not in a function); did you mean ‘Py_TPFLAGS_HAVE_VERSION_TAG’?
  693 |      .tp_flags = (Py_TPFLAGS_DEFAULT | Py_TPFLAGS_HAVE_VECTORCALL
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                        Py_TPFLAGS_HAVE_VERSION_TAG
```


### Runtime information:

Python: Python 3.8.10


### Context for the issue:

_No response_

--- a/setup.py
+++ b/setup.py
@@ -16,8 +16,8 @@ import re
 
 # Python supported version checks. Keep right after stdlib imports to ensure we
 # get a sensible error for older Python versions
-if sys.version_info[:2] < (3, 8):
-    raise RuntimeError("Python version >= 3.8 required.")
+if sys.version_info[:2] < (3, 9):
+    raise RuntimeError("Python version >= 3.9 required.")
 
 
 import versioneer

